docs/design-proposals/multi-tenancy/multi-tenancy-network-mizar-integration.md

```diff
@@ -0,0 +1,201 @@
+* Primary: upon tenant pod creation, automatically attach pod to Mizar VPC/subnet created for this tenant.
+* Primary: Mizar and Arktos decouple, i.e. Mizar manage VPCs and subnets in VPC only, it should not be aware of the ownership
+of different pods and other objects.
+* Secondary: support multiple subnets in same VPC for a single tenant. (Post 2022-01-30)
+* Secondary: support multiple VPC/subnets in same tenant. (Post 2022-01-30)
+
+## 3. Proposal <a name="proposal"></a>
+
+Inject Mizar adapter into Arktos, completely isolated from Mizar management plane and relatively isolated from Arktos network
+management components.
+
+### 3.1. Related Mizar Specs <a name="mizar-spec"></a>
+
+* VPC Spec
+```
+apiVersion: "mizar.com/v1"
+kind: Vpc
+metadata:
+  name: <vpc name>
+spec:
+  ip: "<ip address>"
+  prefix: "<ip address prefix length>"
+  dividers: <number of dividers in vpc>
+  status: "<vpc provisioning status>"
+```
+
+* Subnet Spec
+```
+apiVersion: mizar.com/v1
+kind: Subnet
+metadata:
+  name: <subnet name>
+spec:
+  ip: "<ip address in vpc>"
+  prefix: "<ip address prefix length>"
+  bouncers: <number of bouncers in subnet>
+  vpc: "<vpc name>"
+  status: "<subnet provisioning status>"
+```
+
+* Pod Spec
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: <pod name>
+  annotations:
+    mizar.com/vpc: "<vpc name>"
+    mizar.com/subnet: "<subnet name>"
+```
+
+### 3.2 Related Arktos Specs <a name="arktos-spec"></a>
+* Network Spec
+```
+apiVersion: arktos.futurewei.com/v1
+kind: Network
+metadata:
+  name: <VPC name defined in arktos>
+  tenant: <tenant name>
+spec:
+  type: vpc
+  vpcID: <VPC name in Mizar>
+```
+
+* Pod Spec
+```
+piVersion: v1
+kind: Pod
+metadata:
+  name: <pod name> 
+  tenant: <tenant name>
+  labels:
+    arktos.futurewei.com/network: <VPC name defined in arktos> 
+```
+
+### 3.3 Arktos Pod Creation Workflow <a name="pod-creation-wf"></a>
+#### 3.3.1 Workflow for default VPC and subnet in tenant <a name="pod-creation-default-wf"></a>
+1. Client creates Tenant
+2. Tenant controller creates default network object for tenant
+3. Mizar-arktos-network-controller (existing controller) **creates Mizar VPC and subnet, updates corresponding arktos network object (TODO)**
+   * Target VPC Spec
+```
+# Sample VPC Spec
+
+apiVersion: "mizar.com/v1"
+kind: Vpc
+metadata:
+  name: vpc-tenant-a
+spec:
+  ip: "10.0.0.0"
+  prefix: "8"
+  dividers: 5
+  status: "Init"
+
+# Proposed VPC template
+{
+   "apiVersion": "mizar.com/v1",
+   "kind": "Vpc",
+   "metadata": {
+      "name": "{$vpc_name}"
+   },
+   "spec": {
+      "ip": "{$ip_address}",
+      "prefix": "{$ip_prefix_length}",
+      "dividers": {$num_of_dividers},
+      "status": "Init"
+   }
+}
+```
+
+   * Target Subnet Spec
+```
+# Sample Subnet Spec
+apiVersion: mizar.com/v1
+kind: Subnet
+metadata:
+  name: subnet-tenant-a
+spec:
+  ip: "10.0.0.0"
+  prefix: "16"
+  bouncers: 3
+  vpc: "vpc-tenant-a"
+  status: "Init"
+  
+# Proposed Subnet template
```

Comment:
the implementation would not go with template. Do we need to keep this template in design doc?

Reply:
removed